src/Core/Cache/DefaultCacheFactory.php

```diff
@@ -20,59 +23,101 @@
  */
 class DefaultCacheFactory implements CacheFactory
 {
-
     /**
      * @var string Absolute directory path
      */
-    protected $directory;
+    protected $args = [];
 
     /**
-     * @var string APC version for apcu_add()
+     * @var LoggerInterface
      */
-    protected $version;
+    protected $logger;
 
     /**
-     * @param string $directory
-     * @param string $version
+     * @param array $args List of global options to merge with args during create()
+     * @param LoggerInterface $logger Logger instance to assign
      */
-    public function __construct($directory, $version = null)
+    public function __construct($args = [], LoggerInterface $logger = null)
     {
-        $this->directory = $directory;
-        $this->version = $version;
+        $this->args = $args;
+        $this->logger = $logger;
     }
 
     /**
      * @inheritdoc
      */
     public function create($service, array $args = array())
     {
-        $namespace = (isset($args['namespace'])) ? $args['namespace'] : '';
-        $defaultLifetime = (isset($args['defaultLifetime'])) ? $args['defaultLifetime'] : 0;
-        $version = $this->version;
-        $directory = $this->directory;
+        // merge args with default
+        $args = array_merge($this->args, $args);
+        $namespace = isset($args['namespace']) ? $args['namespace'] : '';
+        $defaultLifetime = isset($args['defaultLifetime']) ? $args['defaultLifetime'] : 0;
+        $directory = isset($args['directory']) ? $args['directory'] : null;
+        $version = isset($args['version']) ? $args['version'] : null;
+
+        // Check support
+        $apcuSupported = $this->isAPCUSupported();
+        $phpFilesSupported = $this->isPHPFilesSupported();
+
+        // If apcu isn't supported, phpfiles is the next best preference
+        if (!$apcuSupported && $phpFilesSupported) {
+            return $this->createCache(PhpFilesCache::class, [$namespace, $defaultLifetime, $directory]);
+        }
```

Comment:
I haven't tested but in PHP 7 you can use PhpArrayAdapter: https://github.com/symfony/cache/blob/master/Adapter/PhpArrayAdapter.php